app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -42,4 +34,27 @@ def confirm_user_authenticated_for_2fa_setup
     return unless MfaPolicy.new(current_user).two_factor_enabled?
     redirect_to user_two_factor_authentication_url
   end
+
+  def user_needs_confirmation_screen?
+    (next_setup_choice.present? ||
+      user_session[:suggest_second_mfa]
+    ) &&
```

Comment:
Stylewise, I'd do this:
```suggestion
    (next_setup_choice.present? || user_session[:suggest_second_mfa]) &&
```